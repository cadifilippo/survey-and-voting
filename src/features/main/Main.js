import { Link } from 'react-router-dom';

import styles from './Main.module.css';

const Main = () => {
  return (
    <section className={styles.content}>
      <h1>App: Survey & Voting</h1>
      <button>
        <Link to="/survey/new">Survey</Link>
      </button>
      <button>
        <Link to="/voting/new">Voting</Link>
      </button>
    </section>
  );
};

export default Main;
