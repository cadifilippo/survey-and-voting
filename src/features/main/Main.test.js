import { expect, test } from '@jest/globals';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Main from './Main';

describe('Main', () => {
  test('render correctly', () => {
    const tree = render(
      <BrowserRouter>
        <Main />
      </BrowserRouter>
    );
    expect(tree).toMatchSnapshot();
  });
});
