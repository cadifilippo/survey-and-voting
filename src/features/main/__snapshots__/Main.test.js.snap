// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="content"
      >
        <h1
          class="title"
        >
          App: Survey & Voting
        </h1>
        <section
          class="buttons"
        >
          <a
            class="button__text"
            href="/survey/new"
          >
            <button
              class="button"
            >
              Survey
            </button>
          </a>
          <a
            class="button__text"
            href="/voting/new"
          >
            <button
              class="button"
            >
              Voting
            </button>
          </a>
        </section>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="content"
    >
      <h1
        class="title"
      >
        App: Survey & Voting
      </h1>
      <section
        class="buttons"
      >
        <a
          class="button__text"
          href="/survey/new"
        >
          <button
            class="button"
          >
            Survey
          </button>
        </a>
        <a
          class="button__text"
          href="/voting/new"
        >
          <button
            class="button"
          >
            Voting
          </button>
        </a>
      </section>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
